consumergroup.go

```diff
@@ -555,7 +555,7 @@ func (g *Generation) partitionWatcher(interval time.Duration, topic string) {
 type coordinator interface {
 	io.Closer
 	findCoordinator(findCoordinatorRequestV0) (findCoordinatorResponseV0, error)
-	joinGroup(joinGroupRequestV1) (joinGroupResponseV1, error)
+	joinGroup(joinGroupRequestV1) (joinGroupResponse, error)
```

Comment:
I think we should update joinGroupRequestV1 to be called joinGroupRequest since it actually is now not necessarily V1